node/node.go

```diff
@@ -292,6 +295,29 @@ func (n *Node) run(ctx context.Context) (err error) {
 				}
 				n.Unlock()
 				renewCert()
+			case newRoot := <-n.notifyRootCAChange:
+				n.Lock()
+				role := n.role
```

Comment:
`notifyNodeChange` is a buffered channel. If the role is changed at the same time as the root CA, this case and the one above might happen in either order. It's possible to skip this root CA change even if we are no longer a manager, or to handle it even if we have just become a manager.

I feel like it would be safer to combine these two channels into a single channel, that passes a struct with a node pointer (which may be nil) and a new root (which may be empty).

I think that on managers, we should only skip the call to `UpdateRootCA`, and not the send on `n.notifyTLSChange`. Otherwise, the agent will continue to report to the dispatcher that it is using the old TLS root. I think there will be less potential for bugs if all nodes report up-to-date information to the dispatcher. It might simplify our logic for deciding when the root rotation is complete.